AsyncDisplayKit/Private/_ASHierarchyChangeSet.mm

```diff
@@ -132,6 +133,31 @@ - (instancetype)initWithOldData:(std::vector<NSInteger>)oldItemCounts
 
 #pragma mark External API
 
+- (void (^)(BOOL finished))completionHandler
+{
+  [self _ensureCompleted];
+
+  void (^completionHandler)(BOOL) = _completionHandler;
+  _completionHandler = nil;
+  return completionHandler;
+}
+
+- (void)addCompletionHandler:(void (^)(BOOL))completion
```

Comment:
The method actually does _add_ a completion handler. I've updated the documentation to explain what's going on.